vector/src/main/java/im/vector/activity/VectorRoomActivity.java

```diff
@@ -1943,7 +1947,12 @@ public void run() {
         }
 
         if (0 != sharedDataItems.size()) {
-            mVectorRoomMediasSender.sendMedias(sharedDataItems);
+            if (sharedDataItems.get(0).getFileName(this).indexOf(".jpg")>-1) {
+                intent.setClass(this, MediasPreviewerActivity.class);
```

Comment:
1- some other image types can be supported (png, gif...)
2- you cannot guess the list items type from the first one : with a files explorer, you could select a file, a video and an image.